### ISSUE TITLE
[BUG] clang++ exited with code 1

### ISSUE THREAD

**Issue by zwikk** [microsoft/vscode-dotnettools#744]
### Describe the Issue

When building a MAUI project on Apple silicon (M1) that references a dependency that doesn't support iossimulator-arm64, the following build error occurs:
```
error : clang++ exited with code 1:
```

The MAUI project needs to be build for runtime **iossimulator-x64**, but there's no option in VSCode to specify this.

Observations:
- ```ForceSimulatorX64ArchitectureInIDE``` has no effect in VSCode
- ```<RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>``` has no effect when set explicitly in csproj
- The attached project compiles and runs fine with VSMac on the same M1 machine
- Compiling with CLI (```dotnet build -c Debug -f net8.0-ios -r iossimulator-x64```) works fine

The following issues are related, but do not contain a solution:
- https://github.com/dotnet/maui/issues/16778
- https://github.com/dotnet/maui/issues/17236

### Steps To Reproduce

1. Clone repo https://github.com/zwikk/maui-playground/tree/main/Source/VsCodeTest
2. Open solution in VSCode
3. Try to build targetting iOS
4. Observe the build error
5. Optional: open solution in VSMac, observe build without errors
7. Optional: build solution with dotnet CLI, observe build without errors
   ```dotnet build -c Debug -f net8.0-ios -r iossimulator-x64```

### Expected Behavior

Be able to build the project just as it works with VSMac and dotnet CLI. 

### Environment Information

MacOS 14.1
VSCode Version: 1.84.2 (Universal)
dotnet 8.0.100
Apple M1 Pro

> VS bug [#1921029](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1921029)

**Comment by mauroa** [comment]
Hi @zwikk, this issue is currently being investigated. Our team will get back to you if either more information is needed, a workaround is available, or the issue is resolved.

**Comment by Syed-RI** [comment]
I have faced similar issue and the linked repo is here: https://github.com/xamarin/xamarin-macios/issues/19596

**Comment by edgedj** [comment]
This issue prevents running the iOS simulator on Apple silicon. At present Visual Studio for Mac knows to compile a maui app with the flag -r iossimulator-x64 but VS Code only compiles with -r iossimulator-arm64 which works fine for Android but iOS won't load as Microsoft App Center and Crash Analytics don't work with arm64 natively. 

Providing an option in the launch.json to tell VS Code to build using iossimulator-x64 would make this possible.

Visual Studio for Mac is disappearing in August next year and we will need a solution by then to carry on developing Maui iOS applications

**Comment by Syed-RI** [comment]
> -r iossimulator-x64

How do one go about and do that for VS Code or any other IDE?

**Comment by edgedj** [comment]
I believe the correct action here is for someone to modify the maui launch configuration to allow us to set "runtime", since we are already able to specify "configuration" and this maps through of to the final botnet build command. launch json would then look like this:

```
        {
            "name": ".NET MAUI",
            "type": "maui",
            "request": "launch",
            "preLaunchTask": "maui: Build",
            "configuration": "Debug",
            "runtime": "iossimulator-x64"
        }
```

**Comment by Syed-RI** [comment]
> I believe the correct action here is for someone to modify the maui launch configuration to allow us to set "runtime", since we are already able to specify "configuration" and this maps through of to the final dotnet build command. launch json would then look like this:
> 
> ```
>         {
>             "name": ".NET MAUI",
>             "type": "maui",
>             "request": "launch",
>             "preLaunchTask": "maui: Build",
>             "configuration": "Debug",
>             "runtime": "iossimulator-x64"
>         }
> ```
<img width="767" alt="Screenshot 2023-12-08 at 15 36 43" src="https://github.com/microsoft/vscode-dotnettools/assets/100694732/662feff4-7ea4-4c6e-b945-41d270ec2314">

Im sorry where and how do I ad these configs? I can only see a config for the Windows Machine above. Sorry im a VS Code noob so please excuse the obvious.

**Comment by Syed-RI** [comment]
@zwikk thanks bud, anyone able to find any workaround yet?

**Comment by emaf** [comment]
This is now fixed in the prerelease channel (v0.8.3). The extension will build using the `RuntimeIdentifier` MSBuild property if present.

**Comment by awattar** [comment]
I'm still hitting this in MS.iOS (non Maui project) because while running in VSCode generated CLI command

`dotnet build -t:Build -p:Configuration=Release -r iossimulator-arm64`

overrides `RuntimeIdentifier` specified in csproj.
